test/libyul/SyntaxTest.cpp

```diff
@@ -0,0 +1,145 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "libyul/AsmAnalysis.h"
+#include "libyul/AsmAnalysisInfo.h"
+#include <libyul/AsmParser.h>
+
+#include "libyul/backends/wasm/WasmDialect.h"
+#include <libyul/backends/evm/EVMDialect.h>
+#include <libyul/backends/wasm/WasmDialect.h>
+
+#include "liblangutil/EVMVersion.h"
+#include <liblangutil/Exceptions.h>
+
+#include <test/libyul/SyntaxTest.h>
+
+using namespace std;
+using namespace solidity;
+using namespace solidity::util;
+using namespace solidity::langutil;
+using namespace solidity::yul::test;
+
+namespace
+{
+std::map<string const, yul::Dialect const& (*)(langutil::EVMVersion)> validDialects = {
+	{
+		"evm",
+		[](langutil::EVMVersion _evmVersion) -> yul::Dialect const&
+		{ return yul::EVMDialect::strictAssemblyForEVM(_evmVersion); }
+	},
+	{
+		"evmTyped",
+		[](langutil::EVMVersion _evmVersion) -> yul::Dialect const&
+		{ return yul::EVMDialectTyped::strictAssemblyForEVM(_evmVersion); }
+	},
+	{
+		"yul",
+		[](langutil::EVMVersion) -> yul::Dialect const&
+		{ return yul::Dialect::yulDeprecated(); }
+	},
+	{
+		"ewasm",
+		[](langutil::EVMVersion) -> yul::Dialect const&
+		{ return yul::WasmDialect::instance(); }
+	}
+};
+
+vector<string> const validDialectNames()
+{
+	vector<string> names{size(validDialects), ""};
+	transform(begin(validDialects), end(validDialects), names.begin(), [](auto const& dialect) { return dialect.first; });
+
+	return names;
+}
+}
+
+void SyntaxTest::parseAndAnalyze()
+{
+	string dialectName = m_validatedSettings.count("Dialect") ? m_validatedSettings["Dialect"] : "evmTyped";
+
+	yul::Dialect const& dialect = validDialects[dialectName](m_evmVersion);
```

Comment:
Maybe better to store this as a member during setting validation?

Reply:
I did it that way first but upon evaluating the `validateSettings()` function further I changed it. Because the description & usage (in other places) of the function says that it has two roles: Make tests not being run (evm version) and move valid settings from `m_settings` to `m_validatedSettings`.

A side effect of this is that we can save a member variable that we don't really need.. which means less code...

Reply:
This will no be properly done with #8244